### DISCUSSION TITLE
Bridgetown + CMS

### DISCUSSION THREAD

**Discussion by jaredmoody** [bridgetownrb/bridgetown#446] [Q&A]
A CMS would be a hurdle I'd have to clear in order to pitch moving my company site to bridgetown. 

Is anybody using a CMS on top of bridgetown like Netlify CMS or something else? What has your experience been?

**Comment by ayushn21** [comment]
@jaredcwhite has put in a lot of work on this particular subject and he's polishing up a plugin for Prismic I believe. I'm sure he can elaborate.

Anyway, I have a number of BT websites that use Netlify CMS and it's ropey to say the least. Nothing to with Bridgetown, just Netlify CMS itself isn't particularly slick and next to impossible to get working outside of Netlify itself; so in all practical terms you're locked into the hosting provider. The only reason I use it is because I couldn't find another free-to-use git based CMS 🤷‍♂️. Not something I'd personally recommend if you had any other options ...

**Comment by jaredcwhite** [comment]
Yeah, I've heard of a few people trying to use Netlify CMS with Bridgetown and it's not super great (and not Bridgetown's fault particularly). I've heard rumblings about an effort at CloudCannon to add support for Bridgetown, but nothing yet to announce.

At this point I feel like for projects that need a robust content authoring solution, a bona fide headless CMS is where it's at. A few months ago I evaluated pretty much every one out there for a client, ended up selecting Prismic, and then subsequent effort on that project has lead to me building an unofficial-official Prismic plugin for Bridgetown. It's not in a widely-released format just yet, but technically it's now available—details at: https://github.com/bridgetownrb/bridgetown-prismic

If you want to kick the tires and let me know how it goes, that'd be cool!

**Reply by jaredcwhite** [reply]
Just want to make sure you got my reply ⬆️ @jaredmoody

**Comment by sandstrom** [comment]
I've found this one: https://spinalcms.com/cms-for-bridgetown/

Haven't tried it, and looks pretty unpolished at this stage, but could be worth looking at.

Sadly they don't support two-way sync, so it seems like they want to be the source of truth. Would have been better if we could have developers edit via Github, and let marketing people edit via Spinal, but have their stuff go through a PR flow on Github (without having to give them access to Github).

**Reply by jaredcwhite** [reply]
I think Spina does use the Git repo as the source of truth. I haven't tried it out yet, but on their site it says: "Spinal connects to your GitHub repo to pull content you have and to push new content you created." and later: "Publish to create a commit to your main branch"

**Reply by sandstrom** [reply]
I talked with someone on the Spina team (to provide feedback along with what I wrote above), seems like they currently may have themselves as the source of truth (somewhat weird, I know), but that they have plans on making Github the source of truth in the future. Not 100% certain about this though.

**Reply by jaredcwhite** [reply]
Ah yeah, I found the FAQ entry where they do say that. Hmm, disappointing. (Also I goofed earlier and called it Spina rather than Spinal…there's actually _another_ CMS called [Spina](https://spinacms.com)).

**Comment by aiomaster** [comment]
Just in case someone is interested.
I am using the headless CMS [Directus](https://directus.io/) to edit the content of my bridgetown site.
I've written a little plugin that triggers bridgetown deployment, when someone changes something in directus.
To get the data to bridgetown I use the bridgetown plugin [graphtown](https://github.com/whitefusionhq/graphtown) that fetches the data via the build in graphql API of directus. I have everything bundled in a docker container and it is working fine for me.

**Comment by ghost** [comment]
Would be interesting to get Ghost working on BridgetownRB.

**Reply by bigtiger** [reply]
Personally, I've found the Ghost data model to be too limiting to be of much use. If you're working with Pages, Posts, and Authors as your core concepts you'll probably be fine. Beyond that I've really struggled to perform some basic CMS tasks like creating a collection of items that are not pages, posts, or authors without having to resort to hacks. That was before version 5 so it's possible that's changed.

**Reply by ghost** [reply]
Yeah, I'm only interested in creating posts and pages via Ghost while I build everything else via BridgetownRB. In the meantime, I may convert to Astro if I can't figure out how to use it with BridgetownRB.

**Comment by ghost** [comment]
Wasn't compatible with a company website I built on BridgetownRB. Posts were put into their own folder to fit their category in the _posts folder.
Not sure if that's been fixed in Spinal.
Not sure if I can test that, as I'm no longer with that company.

Although, I prefer a selfhosted CMS.

**Comment by PedroAugustoRamalhoDuarte** [comment]
I came across an amazing CMS for Next.js called [Outstatic](https://github.com/avitorio/outstatic). It's a git-based CMS that lives directly inside the application, which I think is a brilliant idea. This could be an incredible feature for Bridgetown in the future!
